Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.h

```diff
@@ -120,6 +120,7 @@ class SOAuthorizationSession : public ThreadSafeRefCountedAndCanMakeThreadSafeWe
     RefPtr<API::NavigationAction> m_navigationAction;
     WeakPtr<WebPageProxy> m_page;
     InitiatingAction m_action;
+    bool m_isInDestructor { false };
```

Comment:
cc @pascoej , On Main, we can simply use `RefAllowingPartiallyDestroyed` instead of this data member. It doesn't stop landing this fix but we should consider cleaning up in a follow up.